src/stateMachineFwk/controlBlock.cpp

```diff
@@ -116,11 +128,15 @@ namespace SM
 		}
 	}
 
-	void ControlBlock::setNextState(State* state)
-	{
-		if (tdb_mp != NULL)
-			tdb_mp->setNextState(state);
-	}
+    void ControlBlock::qInternalEvent(Event &event)
+    {
+        std::lock_guard<std::mutex> lock(mutex_m);
```

Comment:
do we need this ? if sinlge thread own cb ?

Reply:
Internal EventQ is used only by state Machine thread. So the locking is not required. Will remove.